app/controllers/pets_controller.rb

```diff
@@ -0,0 +1,27 @@
+class PetsController < ApplicationController
+
+def edit
+  @user = current_user
+  @pet = Pet.find(params[:id])
+end
+
+def show
+  @pet = Pet.find(params[:id])
```

Comment:
Maybe it makes sense to create a method `set_pet` and call it before the methods `edit`, `show` and others if we need?

Reply:
Issues fixed